src/k8s-engine/k3sHelper.ts

```diff
@@ -431,9 +431,8 @@ export default class K3sHelper extends events.EventEmitter {
         console.log('Error verifying checksums after download', error);
         throw error;
       }
-      await safeRename(workDir, path.join(cacheDir, version.raw));
     } finally {
-      await fs.promises.rmdir(workDir, { recursive: true, maxRetries: 3 });
+      await safeRename(workDir, path.join(cacheDir, version.raw));
```

Comment:
```suggestion
      await fs.promises.rm(workDir, { recursive: true, maxRetries: 3, force: true });
```
 and leave the `safeRename` where it was

Reply:
Thanks for suggesting Jan.
Did the refactor, removing the deprecated `rmdir` to `rm` and it's working properly.